components/giftbox.vue

```diff
@@ -0,0 +1,301 @@
+            </div>
+            <p>{{ feature.title }}</p>
+          </div>
+        </div>
+        <div v-if="selectedFeature" class="features-detail">
+          <h6>{{ selectedFeature.title }}</h6>
+          <a
+            v-if="selectedFeature.image"
+            :href="selectedFeature.image"
+            target="_blank"
+          >
+            <img
+              :src="selectedFeature.image"
+              :alt="selectedFeature.title"
+              class="feature-image"
+            />
+          </a>
+          <!-- eslint-disable-next-line vue/no-v-html -->
+          <div class="feature-content" v-html="selectedFeature.body" />
+          <div class="flex-right">
+            <a
+              v-if="selectedFeature.callToAction"
+              :href="selectedFeature.callToAction"
+              target="_blank"
+            >
+              <b-button class="btn-call-to-action" variant="primary">
+                {{ selectedFeature.callToActionLabel }}
+              </b-button>
+            </a>
+          </div>
+        </div>
+        <div v-else class="features-detail">
+          <p>{{ $t('giftbox.no_features') }}</p>
+        </div>
+      </div>
+    </b-modal>
+  </div>
+</template>
+
+<script>
+  export default {
+    name: 'Giftbox',
+    data: () => ({
+      loading: true,
+      features: [],
+      selectedFeature: undefined,
+    }),
+    computed: {
+      seenFeatures() {
+        return this.$cookies.get('seenFeatures') || [];
+      },
+      cookieOptions() {
+        return {
+          maxAge: 60 * 60 * 24 * 30,
+        };
+      },
+    },
+    created() {
+      this.fetchFeatures().then((res) => {
+        this.features = res;
+        // sync the seenFeatures with the incoming features from the api call
+        this.syncSeenFeatures();
+        // if there are new features
+        if (this.features.length > 0) {
+          this.selectedFeature = this.features[0];
+          this.addToSeenFeatures(this.selectedFeature.uid);
+        }
+      });
+    },
+    methods: {
+      async fetchFeatures() {
+        this.loading = true;
+        const response = await fetch(process.env.newFeaturesUrl);
+        const { data } = await response.json();
+        this.loading = false;
+        return data;
+      },
+      addToSeenFeatures(uid) {
+        const seenFeatures = this.seenFeatures;
+        if (!seenFeatures.includes(uid)) {
+          seenFeatures.push(uid);
+          this.$cookies.set('seenFeatures', seenFeatures, this.cookieOptions);
+        }
+      },
+      isFeatureSeen(uid) {
+        return this.seenFeatures.includes(uid);
+      },
+      syncSeenFeatures() {
+        const seenFeatures = this.seenFeatures;
+
+        if (!this.features.length > 0) {
+          this.resetSeenFeatures();
+        } else {
+          seenFeatures.filter((seenFeature) => {
+            return this.features.find((feature) => feature.uid === seenFeature);
+          });
+        }
```

Comment:
I'm a bit confused by this `if/else` at this point.

If there are no features at all, it calls `resetSeenFeatures()` which sets the cookie value to `[]`. But then at the end of this method (`syncSeenFeatures`) it also sets the cookie value to `seenFeatures`, which is also `[]` in that case. So `resetSeenFeatures` seems unnecessary.

So then it could be re-written as:
```
        if (this.features.length > 0) {
          seenFeatures.filter((seenFeature) => {
            return this.features.find((feature) => feature.uid === seenFeature);
          });
        }
```

However, the code in that new `if` block (originally the `else`) also does nothing.

1. It doesn't do anything with the return value of `.filter()` (which returns a new array)
2. It will never filter anything anyway, because you're checking that a feature in `seenFeatures` is also in `this.seenFeatures`. But those two arrays are exactly the same, because `const seenFeatures = this.seenFeatures`

So that code can also be removed, and you end up with:

```
syncSeenFeatures() {
  this.$cookies.set('seenFeatures', seenFeatures, this.cookieOptions);
}
```

Unless I'm mistaken/overlooking something?

But if it's only job is storing the seenFeatures in the cookie, maybe it should be called `persistSeenFeatures` instead of `syncSeenFeatures`.

Reply:
Additionally, if you simplify it to `persistSeenFeatures` as suggested, you can also use it in `addToSeenFeatures` instead of doing `this.$cookies.set()` there.

Reply:
Ah, I was confusing `this.seenFeatures` with `this.features`!

Reply:
Still, the filter does nothing right now and you should assign the return value to something 😉

Reply:
```
seenFeatures.filter((seenFeature) => {
            return this.features.find((feature) => feature.uid === seenFeature);
});
```
 has to be reassigned to seenFeatures